src/scales/scale.time.js

```diff
@@ -752,12 +752,19 @@ module.exports = Scale.extend({
 	 */
 	getLabelCapacity: function(exampleTime) {
 		var me = this;
+		var margins = me.margins;
 
 		// pick the longest format (milliseconds) for guestimation
 		var format = me.options.time.displayFormats.millisecond;
 		var exampleLabel = me.tickFormatFunction(exampleTime, 0, [], format);
 		var tickLabelWidth = me.getLabelWidth(exampleLabel);
-		var innerWidth = me.isHorizontal() ? me.width : me.height;
+
+		// Using margins instead of padding because padding is not calculated
+		// at this point (buildTicks). Margins are provided from previous calculation
+		// in layout steps 5/6
+		var innerWidth = me.isHorizontal()
```

Comment:
This calculation is confusing to me. If I'm thinking about margin and padding as it relates to the CSS box model then the width would not include the margin and so it wouldn't make sense to subtract the margin from the width. It would be helpful to add some documentation to `margins` explaining what exactly the variable represents and how it is used.

Reply:
I'm not going to document margins in this PR.

Reply:
Ok. This still doesn't make sense to me though. The margins are not included in the width, right? It seems like maybe you're just getting lucky that the margins and padding happen to have similar values in the case that you're testing.

Reply:
Scale is updated 2 times during layout. First to determine its minimum dimensions (height for horizontal and width for vertical scale)

This first update is without margins. After update scale is asked for padding. Now scale has generated ticks (with 0 margin), and padding includes approximately half of the tick label (if ticks are visible), each side.

2nd update is with margins. Those margins include the padding from previous update and possible other scales in the side (max value of the padding / scale size).  Top and left boxes are placed first, so bottom box will have left box width in margin if its greater than its own padding calculation.

This 2nd update is what is fixed here.

Finally box left, top, right and bottom are set, but box is not updated 3rd time. Margins are subtracted from these values. So in any possible later updates, margins are not provided, but width has already been adjusted.

If it still does not make sense, please read / debug here:
https://github.com/chartjs/Chart.js/blob/ddee91eb9f7f68d64b1634c103ebbc8421b23c7a/src/core/core.layouts.js#L149